application/core/LimeMailer.php

```diff
@@ -219,14 +233,10 @@ public static function getInstance($reset=self::ResetBase)
         }
         /* Some part must be always resetted */
         if ($reset) {
-            self::$instance->clearAddresses(); // Unset only $this->to recepient
-            self::$instance->clearAttachments(); // Unset attachments (maybe only under condition ?)
-            self::$instance->oToken = null;
-            self::$instance->eventName = 'beforeEmail';
+            self::$instance->init();
             if(self::$instance->surveyId) {
-                self::$instance->eventName = 'beforeSurveyEmail';
```

Comment:
No event name?

Reply:
`self::$instance->setSurvey(self::$instance->surveyId);` set the eventName (and reset html on/off)

Reply:
Alright.

The method setSurvey has description "Set email for this survey". Traditionally, a setter method has very little logic. Maybe we should rename that method in the future?

Will merge.

Reply:
Yep , right.

But it's not setSurveyId :) what name you propose ?

Reply:
Maybe "configureSurveyEmail"?

Reply:
More configureSurvey , we are aready in mailer …
API update or not ?

Reply:
Bah. Dunno.

Reply:
If the mailer is never used without a survey, the survey object should be passed in the constructor.

Reply:
LimeMailer is used for ALL mail , not related to survey too …

See the feature request about event for mailer …

Reply:
Alright.

Reply:
https://github.com/LimeSurvey/LimeSurvey/pull/1246

> New feature #09668: attachment in beforeTokenEmail
> Fixed issue #12243: beforeTokenEmail didn't happen for remote control
> Fixed issue #10529: Unable to send email via some smtp server
> New feature : allow to add cc, bcc to any email.

And some other 
New feature #14622: allow to update any email text : 
- https://github.com/LimeSurvey/LimeSurvey/pull/1246/commits/6398e9685633ecc21880777093739740ba721d9c
- […]
- https://github.com/LimeSurvey/LimeSurvey/pull/1246/commits/0f63a9cdc9da13cd34e46e25a373a8d0d25ab12c

Reply:
About new feature, we'll have a meeting this week to discuss release process. I'll get back to you after that.

Reply:
@olleharstedt : here it's new feature done **before** 4.0.0RC1 … then … …

Reply:
Old PRs?

Reply:
Yes, the feature with LimeMailer are done in this https://github.com/LimeSurvey/LimeSurvey/pull/1246 pull request